### PR TITLE
feat!: support prettier v3

### LINKED ISSUE #164: pretty-quick will break with prettier v3
Not an issue right now, but pretty-quick will break with v3 due to the removal of resolveConfig.sync (https://github.com/prettier/prettier/pull/12788).

### PR DESCRIPTION
**Pull request by JounQin**
close #164

Waiting https://github.com/prettier/pretty-quick/pull/180#issuecomment-1894971802

cc @SukkaW

### CODE REVIEW

**Comment by SukkaW** [comment]
@JounQin What about bumping the minimum Node.js version as well? Currently `pretty-quick` supports `Node.js >= 10`.

**Comment by JounQin** [comment]
> @JounQin What about bumping the minimum Node.js version as well? Currently `pretty-quick` supports `Node.js >= 10`.

I've already bumped to `>=14`.

File: package.json [hunk 1]
```diff
@@ -35,7 +44,7 @@
   ],
   "scripts": {
     "build": "run-p 'build:*'",
-    "build:r": "r -f esm -o dist || exit 0",
+    "build:r": "r -f esm || exit 0",
```
Comment (JounQin) [inline package.json:47]:
```suggestion
    "build:r": "r -f esm",
```

**Comment by taylorkline** [comment]
Should prettier/prettier#15337 be reverted and `pretty-quick` be added back to the docs of pre-commit solutions?

**Pull request merged**